### PR TITLE
[MIG][10.0] stock account operating unit

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Stock account moves with Operating Unit
=======================================

This module introduces the following features:
- Creates account move lines when stock moves are posted between internal
locations within the same company, but different OU’s.

The PR in v9: https://github.com/OCA/operating-unit/pull/21

### CODE REVIEW

**Comment by AaronHForgeFlow** [comment]
cc @eficent

File: stock_account_operating_unit/README.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+Configuration
+=============
+
+If your company is required to generate a balanced balance sheet by
+operating unit you can specify at company level that operating units should
+be self-balanced, and then indicate a self-balancing clearing account.
+
+* Create an account for "Inter-OU Clearing" of type Regular.
+* Go to *Settings / Companies / Configuration* and:
+   ** Set the "Operating Units are self-balanced" checkbox
```
Comment (elicoidal) [inline stock_account_operating_unit/README.rst]: insert a blank line

File: stock_account_operating_unit/README.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+
+Configuration
+=============
+
+If your company is required to generate a balanced balance sheet by
+operating unit you can specify at company level that operating units should
+be self-balanced, and then indicate a self-balancing clearing account.
+
+* Create an account for "Inter-OU Clearing" of type Regular.
```
Comment (elicoidal) [inline stock_account_operating_unit/README.rst]: use better "#. " for enumerated list

**Review by elicoidal — APPROVED** [review]
LGTM (no functional tests)

**Comment by AaronHForgeFlow** [comment]
Thank you for your review @elicoidal I applied the changes you suggested in the readme

**Review by elicoidal — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
@guewen Can you take a look at this? This module is basic for the operating unit repo, it would control the stock movements between operating units.

File: stock_account_operating_unit/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    "name": "Stock account moves with Operating Unit",
+    "summary": "Create journal entries in moves between internal locations "
+               "with different operating units.",
+    "version": "10.0.1.0.0",
+    "category": "Generic Modules/Sales & Purchases",
+    "author": "Eficent Business and IT Consulting Services S.L., "
+              "Serpent Consulting Services Pvt. Ltd.,"
+              "Odoo Community Association (OCA)",
+    "license": "LGPL-3",
+    "website": "https://github.com/OCA/operating-unit.com",
```
Comment (MiquelRForgeFlow) [inline stock_account_operating_unit/__manifest__.py]: remove the <code>.com</code>

**Review by MiquelRForgeFlow — CHANGES_REQUESTED** [review]

**Comment by AaronHForgeFlow** [comment]
Hi @mreficent your comments have been attended :wink:

**Review by MiquelRForgeFlow — APPROVED** [review]

File: stock_account_operating_unit/README.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
+.. image:: https://img.shields.io/badge/license-LGPLv3-blue.svg
+   :target: https://www.gnu.org/licenses/lgpl.html
+   :alt: License: LGPL-3
+
+=======================================
+Stock account moves with Operating Unit
+=======================================
+
+This module introduces the following features:
+- Creates account move lines when stock moves are posted between internal
```
Comment (LoisRForgeFlow) [inline stock_account_operating_unit/README.rst]: This is not being displayed correctly - Use a blank line before this line

File: stock_account_operating_unit/model/stock_quant.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+from openerp import _, api, exceptions, models
+
+
+class StockQuant(models.Model):
+    _inherit = 'stock.quant'
+
+    @api.multi
+    def _account_entry_move(self, move):
+        """
+        Generate an accounting moves if the product being moved is subject
```
Comment (LoisRForgeFlow) [inline stock_account_operating_unit/model/stock_quant.py]: Generate ~an~ accounting moves (...)

File: stock_account_operating_unit/model/stock_quant.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                          'unit locations is only allowed when both source '
+                          'and destination locations are internal.')
+                    )
+                src_company_ctx = dict(
+                    force_company=move.location_id.company_id.id
+                )
+                company_ctx = dict(company_id=move.company_id.id)
+                self = self.with_context(src_company_ctx)
+                data = move._get_accounting_data_for_valuation()
+                (journal_id, acc_src, acc_dest, acc_valuation) = data
```
Comment (LoisRForgeFlow) [inline stock_account_operating_unit/model/stock_quant.py]: `(journal_id, acc_src, acc_dest, acc_valuation) = move._get_accounting_data_for_valuation()`? So you don't use the intermediate variable `data`?

**Review by LoisRForgeFlow — COMMENTED** [review]
Some comments in line. Plus, to add some test for the UserError would be nice.

**Comment by AaronHForgeFlow** [comment]
@lreficent Thank you for your review. I just realized that the check was not needed because is not possible to create a not internal locations: https://github.com/OCA/operating-unit/blob/10.0/stock_operating_unit/model/stock.py#L78 The UserError will never be triggered so cannot be tested

**Comment by LoisRForgeFlow** [comment]
@aheficent great, this is good from my side now!

**Review by LoisRForgeFlow — APPROVED** [review]

**Pull request merged**